### ISSUE TITLE
Intermittent failure on Postgres tests

### ISSUE THREAD

**Issue by dhensby** [silverstripe/silverstripe-framework#8041]
## Affected Version

3.x and 4.0.x

## Description

The postgres test suite fails intermittently with:

```
There was 1 failure:

1) DataObjectDuplicationTest::testDuplicateManyManyClasses
Match between relation of copy and the original
Failed asserting that 2 matches expected 1.

/home/travis/builds/ss/framework/tests/model/DataObjectDuplicationTest.php:115
/home/travis/builds/ss/vendor/phpunit/phpunit/phpunit:52

FAILURES!
Tests: 1793, Assertions: 7348, Failures: 1, Skipped: 31, Incomplete: 6.
```

## Steps to Reproduce

Run the travis test suite

**Comment by robbieaverill** [comment]
I've added the v4 label since this has made its way into the 4.0 branch now too.

**Comment by dhensby** [comment]
@robbieaverill I think it was introduced in this PR https://github.com/silverstripe/silverstripe-framework/pull/8009#issuecomment-382304116

**Comment by dhensby** [comment]
OK - so it's the typical postgres sorting issue, of course. Which is not helped by the fact that `DataObjectDuplicationTest::testDuplicateManyManyClasses` disguises some of the side-effects of duplication of `ManyMany` relations. Notably how a duplication of a `manymany` on both sides will mean that the relations double up. This means that there are 2 of each `ManyMany` relation after the duplication and so the `First` item of the `ones()` relation is not guaranteed to be the one we are checking against (unless we sort appropriately). see fix at #8132
